<launch>
  <!-- Arguments -->
  <arg name="temporal_updates_on" default="false"/>

  <!-- remote PC -->
  <machine name="remote_pc" address="192.168.1.9" env-loader="/opt/ros/melodic/env.sh" user="milica" default="never"/>

  <!-- sherlock_controller -->
  <node pkg="sherlock" type="sherlock_controller.py" name="sherlock_controller" output="screen" required="true">
    <rosparam file="$(find sherlock)/config/desired_path.yaml" command="load" />
  </node>

  <!-- gmapping -->
  <group unless="$(arg temporal_updates_on)"> 
    <include file="$(find sherlock_slam)/launch/sherlock_slam.launch"/>
  </group>

  <group if="$(arg temporal_updates_on)"> 
    <include file="$(find sherlock_slam)/launch/sherlock_slam.launch">
      <arg name="params_file" value="$(find sherlock_slam)/config/temporal_updates_on.yaml"/>
    </include>
  </group>

  <!-- move_base -->
  <include file="$(find sherlock_navigation)/launch/sherlock_navigation.launch"/>

  <!-- find_object_2d -->
  <include file="$(find sherlock_obj_recognition)/launch/sherlock_object_recognition_remote.launch">
    <arg name="remote_machine_name" value="remote_pc"/>
  </include>
  
</launch>
